src/main/java/com/hermesworld/ais/galapagos/staging/impl/StagingImpl.java

```diff
@@ -220,6 +222,24 @@ private static ApplicableChange checkForCreateTopic(TopicMetadata topic, String
                     newTopic.isSubscriptionApprovalRequired()));
         }
 
+        if (!oldTopic.getProducers().equals(newTopic.getProducers())) {
+            List<String> producerIdsToBeAdded = newTopic.getProducers().stream()
+                    .filter(producer -> !oldTopic.getProducers().contains(producer)).collect(Collectors.toList());
+
+            if (!producerIdsToBeAdded.isEmpty()) {
+                result.add(ChangeBase.TopicProducerAddChange(newTopic.getName(), producerIdsToBeAdded));
```

Comment:
At this position, I must give the general advice to also change the "addProducer" logic to only accept one producer at a time - everywhere. Then, here, you would add one "TopicProducerAddChange" Change for **every** added producer. Imagine that a user added a producer on one day, and serveral days later, a second one. Then, only the first shall be staged to the next stage. This would be impossible with this implementation.